### ISSUE TITLE
Getting HTTP/2 stream 0 was not closed cleanly

### ISSUE THREAD

**Issue by ruske88** [jcmoraisjr/haproxy-ingress#642]
Hi, one of our pod got restarted few days ago and it started with the latest tag, which means it is running haproxy 1.9.16(let's call it POD-new) Other pods are on previous haproxy version, 1.9.15 (let's call it POD-old)
our DNS record was pointing some URL to kubernetes node which run POD-new. Since  this pod restart, we are getting
this error: "curl: (92) HTTP/2 stream 0 was not closed cleanly: Unknown error code (err 2)". Once we changed DNS to point to some kubernetes node which run POD-old, our same request working as expected.
we are using image `quay.io/jcmoraisjr/haproxy-ingress` with args: 
```
--default-backend-service=$(POD_NAMESPACE)/ingress-default-backend
--configmap=$(POD_NAMESPACE)/haproxy-ingress
--sort-backends 
``` 
where configmap is empty, meaning only default settings applied.
Our current k8s ingress definition that is bind to DNS record mentioned above doesn't have any additional annotations, just TLS and backend paths defined.

Is there something that we need to change either on ingress-controller, on add some specific annotation when creating new k8s ingress in order to fix this error above? 
Thanks!

**Comment by jcmoraisjr** [comment]
Hi, I'm not aware of a controller config which would change the behavior in such a way, it seems to me that it's a behavior changed in the haproxy itself.

Perhaps you can give v0.10 a chance? It's still beta but good enough, we're using in production since the second snapshot.

Regarding v0.9, haproxy 1.9 is eol now and I don't think they'll emit a new release, but we could ask. Can you reproduce this behavior using haproxy without the controller? It always happen? Please share the reproducible steps if you have them.

**Comment by ruske88** [comment]
Hi, thanks for the answer. Unfortunately, I don't have an example to send you to reproduce it. We change image to v0.9 tag, and all services are behaving correctly now. 
However, maybe this issue is related: https://github.com/haproxy/haproxy/issues/798 As far as I can tell, fix that was implemented was merged in haproxy 1.8 and 2.0, but not in 1.9.
You are free to close to this issue.

**Comment by jcmoraisjr** [comment]
Thank you so much for the report and the issue link. Let's leave this issue open, I'm planning to fix this, either reverting back to 1.9.15 or forking and patching 1.9.16.

**Comment by if-kenn** [comment]
Sorry for multiple posts, but thought I should put this on an issue that was open.

Is the change for 1.8 going to be released anytime soon? I see recent releases for 2.0, 2.1, 2.2 & 2.3. It appears that 1.8 has not had a release since before the fix was added. 

https://github.com/haproxy/haproxy/issues/798#issuecomment-669501946

https://github.com/haproxy/haproxy/issues/768#issuecomment-669501543

**Comment by jcmoraisjr** [comment]
As expected 1.9 was never released past 1.9.16, I didn't update the embedded version in the controller and now the controller itself was moved to critical fixes only. We have now controller v0.10 that's also compatible with old k8s versions and it's still being maintained, so the suggestion here is to update the controller version. Closing.
